.github/actions/get-ado-token/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Get Azure DevOps Access Token
+inputs:
+  client-id:
+    description: "The client ID of the application calling Azure DevOps"
+    required: true
+  tenant-id:
+    description: "The tenant ID of the application calling Azure DevOps"
+    required: true
+  organization:
+    description: "The Azure DevOps organization to authenticate with"
+    required: true
+outputs:
+  token:
+    description: "The access token to authenticate with Azure DevOps"
+    value: ${{ steps.ADOAuth.outputs.token }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: OIDC Login with AzPowershell
+      uses: azure/login@v2
+      with:
+        client-id: ${{ inputs.client-id }}
+        tenant-id: ${{ inputs.tenant-id }}
+        allow-no-subscriptions: true
+        enable-AzPSSession: true
+    - id: ADOAuth
+      name: Get ADO Access Token
+      uses: azure/powershell@v1
+      with:
+        azPSVersion: "latest"
+        inlineScript: |
+          $accessToken = (Get-AzAccessToken -ResourceUrl "https://${{inputs.organization}}.visualstudio.com").Token
+          "token=$accessToken" | Out-File -FilePath $env:GITHUB_OUTPUT -Append
```

Comment:
Out of curiosity, is there another mode called `-Write`? Why do we want to do an `-Append` here?